pkg/scheduler/factory_test.go

```diff
@@ -402,9 +420,10 @@ func TestCreateFromConfig(t *testing.T) {
 					}
 				}),
 			)
-
 			if err != nil {
-				t.Fatalf("Error constructing: %v", err)
+				if !strings.Contains(err.Error(), tc.wantErr) {
```

Comment:
Not the best practice to compare error strings.

Please do a follow up where you use `cmp.Equal`